flytekit/core/interface.py

```diff
@@ -259,12 +259,16 @@ def transform_interface_to_list_interface(interface: Interface) -> Interface:
 def _change_unrecognized_type_to_pickle(t: Type[T]) -> Type[T]:
     try:
         if hasattr(t, "__origin__") and hasattr(t, "__args__"):
-            if t.__origin__ == list:
+            if get_origin(t) == list:
                 return typing.List[_change_unrecognized_type_to_pickle(t.__args__[0])]
-            elif t.__origin__ == dict and t.__args__[0] == str:
+            elif get_origin(t) == dict and t.__args__[0] == str:
                 return typing.Dict[str, _change_unrecognized_type_to_pickle(t.__args__[1])]
-        else:
-            TypeEngine.get_transformer(t)
+            elif get_origin(t) == typing.Union:
```

Comment:
can we use `is` instead of == or does that not work here?

Reply:
updated it.